File: src/infra/SessionsRepository.js
var _ = require('lodash');

var SessionsRepository = function SessionsRepository(){
    var self = this;

    var projections = [];

    self.getUserIdOfSession = function getUserIdOfSession(sessionId){
        var projection = _.find(projections, function(projection) {
            return projection.sessionId === sessionId;
        });

        return projection ? projection.userId : null;
    };

    self.save = function save(projection) {
        projections.push(projection);
    };
};

exports.create = function create(){
    return new SessionsRepository();
};

File: Mixter.Infrastructure/SessionsRepository.cs
using System.Collections.Generic;
using Mixter.Domain.Identity;

namespace Mixter.Infrastructure
{
    public class SessionsRepository
    {
        private readonly IDictionary<SessionId, SessionProjection> _projectionsById = new Dictionary<SessionId, SessionProjection>();

        public void Save(SessionProjection projection)
        {
            _projectionsById.Add(projection.SessionId, projection);
        }

        public UserId? GetUserIdOfSession(SessionId sessionId)
        {
            if (!_projectionsById.ContainsKey(sessionId))
            {
                return null;
            }

            var projectionOfSession = _projectionsById[sessionId];
            return projectionOfSession.SessionState == SessionState.Enabled
                ? projectionOfSession.UserId
                : (UserId?)null;
        }
    }
}
